base/linalg/lu.jl

```diff
@@ -179,36 +179,13 @@ function logabsdet{T,S}(A::LU{T,S})  # return log(abs(det)) and sign(det)
         end
         abs_det += log(abs(dg_ii))
     end
-    s = (isodd(c) ? -one(real(T)) : one(real(T))) * P
+    s = (isodd(c) ? -one(T) : one(T)) * P
```

Comment:
Same here. It would also be great if you could change this to the `ifelse` function, i.e. `ifelse(isodd(c), -one(real(T)), one(real(T)))`.